tensorboard/plugins/histogram/metadata.py

```diff
@@ -53,6 +52,19 @@ def parse_plugin_metadata(content):
       corresponding to the histogram plugin.
 
   Returns:
-    A `HistogramMetadata` instance.
+    A `HistogramPluginData` protobuf object.
   """
-  return HistogramMetadata(**json.loads(content))
+  if content == '{}':
+    # Old-style JSON format. Equivalent to an all-default proto.
+    return plugin_data_pb2.HistogramPluginData()
+  else:
+    result = plugin_data_pb2.HistogramPluginData()
+    result.ParseFromString(tf.compat.as_bytes(content))
```

Comment:
My recommendation would be throw ValueError if it isn't a byte string. I think those tf.compat methods are good when you want to say, I don't care if it's UTF8 or unicode, but probably not for legit binary parameters.